server/cluster/cluster.go

```diff
@@ -1020,18 +1022,25 @@ func (c *RaftCluster) RemoveStore(storeID uint64) error {
 		return errs.ErrStoreNotFound.FastGenByArgs(storeID)
 	}
 
-	// Remove an offline store should be OK, nothing to do.
-	if store.IsOffline() {
-		return nil
-	}
-
 	if store.IsTombstone() {
 		return errs.ErrStoreTombstone.FastGenByArgs(storeID)
 	}
 
-	newStore := store.Clone(core.SetStoreState(metapb.StoreState_Offline))
+	if !physicallyDestroyed {
+		if store.IsPhysicallyDestoryAndOffline() {
+			return errors.Errorf("The store is already physically destroyed and offline")
+		}
+		// Remove an offline store should be OK, nothing to do.
+		if store.IsOffline() {
+			return nil
+		}
+	} else if store.IsPhysicallyDestoryAndOffline() {
+		return nil
+	}
+	newStore := store.Clone(core.OfflineStore(physicallyDestroyed))
```

Comment:
prefer to use SetStoreState

Reply:
`SetStoreState` do not have args `physicallyDestroyed` since only offline a store can use the param